### PR TITLE
Dev dcps info report discovery

### PR DESCRIPTION
**Pull request by swolferDF**
- change init_opendds with config args 
-  change on init_opendds_impl for default_rtps config 
- shutdown particpant factory when there are no more particpants.

### CODE REVIEW

File: pyopendds/DomainParticipant.py [hunk 1]
```diff
@@ -10,6 +11,7 @@ def participant_cleanup(*args):
     pass
 
 
+
```
Comment (Sdpierret) [inline pyopendds/DomainParticipant.py:14]: useless line added

File: pyopendds/DomainParticipant.py [hunk 2]
```diff
@@ -35,3 +39,6 @@ def create_subscriber(self, qos=None, listener=None) -> Subscriber:
 
     def create_publisher(self, qos=None) -> Publisher:
         return Publisher(self, qos)
+
```
Comment (Sdpierret) [inline pyopendds/DomainParticipant.py:42]: remove trailing line

File: pyopendds/init_opendds.py [hunk 3]
```diff
@@ -19,12 +19,14 @@ def init_opendds(*args, default_rtps=True, opendds_debug_level=0):
     verbose). It is printed to stdout.
     """
 
-    args = list(sys.argv[1:])
+
+
+    # args = list(sys.argv[1:])
```
Comment (Sdpierret) [inline pyopendds/init_opendds.py:24]: remove if unused

File: tests/basic_test/publisher.py [hunk 4]
```diff
@@ -14,7 +14,12 @@
         init_opendds(opendds_debug_level=1)
  
         domain = DomainParticipant(34)
+
+        
+        # print(domain)
```
Comment (Sdpierret) [inline tests/basic_test/publisher.py:19]: remove if unused

**Review by Sdpierret — CHANGES_REQUESTED** [review]
some cleanup do do

File: tests/basic_test/publisher.py (hunk 4 above)
Comment (swolferDF) [reply tests/basic_test/publisher.py]: done

File: pyopendds/DomainParticipant.py [hunk 5]
```diff
@@ -1,15 +1,14 @@
...
 from .Publisher import Publisher
+from enum import IntEnum
 
 try:
     from _pyopendds import participant_cleanup  # noqa
 except ImportError as e:
     def participant_cleanup(*args):
         pass
     pass
-
```
Comment (Sdpierret) [inline pyopendds/DomainParticipant.py:11]: python format standard : 2 lines before a class definition

File: pyopendds/DomainParticipant.py [hunk 6]
```diff
@@ -34,4 +35,4 @@ def create_subscriber(self, qos=None, listener=None) -> Subscriber:
         return Subscriber(self, qos, listener)
 
     def create_publisher(self, qos=None) -> Publisher:
-        return Publisher(self, qos)
+        return Publisher(self, qos)
```
Comment (Sdpierret) [inline pyopendds/DomainParticipant.py:38]: no new line ad end of file

File: pyopendds/ext/_pyopendds.cpp [hunk 7]
```diff
@@ -244,12 +250,21 @@ PyObject* participant_cleanup(PyObject* self, PyObject* args)
     // Get DomainParticipant_var
     DDS::DomainParticipant* participant =
         get_capsule<DDS::DomainParticipant>(*pyparticipant);
-
-    if (!participant) return nullptr;
-
+    
+    if (!participant){
```
Comment (Sdpierret) [inline pyopendds/ext/_pyopendds.cpp:254]: why reformat
indentation missing
please keep old format

**Review by Sdpierret — CHANGES_REQUESTED** [review]
mainly to avoid adding more "unwanted" change in case of push to mainline repository

File: pyopendds/DomainParticipant.py (hunk 5 above)
Comment (swolferDF) [reply pyopendds/DomainParticipant.py]: done

File: pyopendds/DomainParticipant.py (hunk 6 above)
Comment (swolferDF) [reply pyopendds/DomainParticipant.py]: done

File: pyopendds/ext/_pyopendds.cpp (hunk 7 above)
Comment (swolferDF) [reply pyopendds/ext/_pyopendds.cpp]: done

**Review by Sdpierret — APPROVED** [review]

**Comment by Sdpierret** [comment]
please resolve the conflit before merge

**Pull request merged**